### ISSUE TITLE
checkout branch before pull it

### ISSUE THREAD

**Issue by jjyr** [RetricSu/godwoken-kicker#81]
https://github.com/RetricSu/godwoken-kicker/blob/740b588b2dfdd7eaa5e0760b0c4de2016494e16e/gw_util.sh#L254

My remote branch is deleted, then the command is crashed at this line. We should checkout to the target branch before the git pull.

**Comment by RetricSu** [comment]
> https://github.com/RetricSu/godwoken-kicker/blob/740b588b2dfdd7eaa5e0760b0c4de2016494e16e/gw_util.sh#L254

this code is only used in previous version. 

For now, in newest develop branch, if local package exits, kicker won't try to fetch the remote repo.

https://github.com/RetricSu/godwoken-kicker/blob/3b5e81cbb3a7d6626a20a948f51638ce3fe47998/gw_util.sh#L574-L575

 However, if you set [always_fetch_new_package](https://github.com/RetricSu/godwoken-kicker/blob/3b5e81cbb3a7d6626a20a948f51638ce3fe47998/docker/.build.mode.env#L32) to true (default is false), and if your remote branch is deleted, this might happend. 
 
 But given that the `always_fetch_new_package` is set to true, I think it should be an expected result.
 
----

 TL;DR
 
 version after 3b5e81cbb3a7d6626a20a948f51638ce3fe47998 commit should not worry this issue.
